### ISSUE TITLE
Add Transform stream class

### ISSUE THREAD

**Issue by creatorrr** [mojo-js/obj-stream#2]
Although, it's trivial to implement a `Transform` stream class using `Duplex`, it'd be great to have it bundled as a part of the library. Thoughts? Happy to make a PR if needed.

**Comment by crcn** [comment]
yeah - a Transform stream sounds like a great idea. Feel free to make a PR!

**Comment by creatorrr** [comment]
Hey @crcn, apologies for the late reply. I have been caught up with work and will definitely get to this soon. If someone is really keen for this PR, please ping me here and I'll find time to add it.
